apps/prairielearn/src/models/enrollment.sql

```diff
@@ -31,11 +42,52 @@ WHERE
   user_id = $user_id
   AND course_instance_id = $course_instance_id;
 
--- BLOCK select_enrollment_for_user_in_course_instance_by_pending_uid
+-- BLOCK select_enrollment_by_id
 SELECT
   *
 FROM
   enrollments
 WHERE
+  id = $id;
+
+-- BLOCK select_enrollment_by_uid
+SELECT
+  to_jsonb(e) AS enrollment
+FROM
+  enrollments AS e
+  LEFT JOIN users AS u ON (u.user_id = e.user_id)
+WHERE
+  e.course_instance_id = $course_instance_id
+  AND (
+    e.pending_uid = $uid
+    OR u.uid = $uid
+  );
+
+-- BLOCK invite_existing_enrollment
+UPDATE enrollments
+SET
+  status = 'invited',
+  user_id = NULL,
   pending_uid = $pending_uid
-  AND course_instance_id = $course_instance_id;
+WHERE
+  -- status sanity checks should be performed upstream.
```

Comment:
What sanity checks? IMO either be explicit or remove this comment, this doesn't mean much in isolation.

Reply:
Removed in c0add6bacf9a32aed633d9d75e2ffd537bfce788.